core/src/main/java/io/questdb/cairo/frm/file/ContiguousFileVarFrameColumn.java

```diff
@@ -97,16 +98,41 @@ public void append(long offset, FrameColumn sourceColumn, long sourceLo, long so
                 assert copySize > 0 && copySize < 1L << 40;
 
                 TableUtils.allocateDiskSpaceToPage(ff, varFd, varOffset + copySize);
-                ff.fadvise(sourceColumn.getPrimaryFd(), varSrcOffset, copySize, Files.POSIX_FADV_SEQUENTIAL);
-                if (ff.copyData(sourceColumn.getPrimaryFd(), varFd, varSrcOffset, varOffset, copySize) != copySize) {
-                    throw CairoException.critical(ff.errno()).put("Cannot copy data [fd=").put(varFd)
-                            .put(", destOffset=").put(varOffset)
-                            .put(", size=").put(copySize)
-                            .put(", fileSize=").put(ff.length(varFd))
-                            .put(", srcFd=").put(sourceColumn.getPrimaryFd())
-                            .put(", srcOffset=").put(varSrcOffset)
-                            .put(", srcFileSize=").put(ff.length(sourceColumn.getPrimaryFd()))
-                            .put(']');
+                if (mixedIOFlag) {
+                    if (ff.copyData(sourceColumn.getPrimaryFd(), varFd, varSrcOffset, varOffset, copySize) != copySize) {
```

Comment:
I tried to do the same but I started to do a util method to convert `ff.coptData` into mmap + memcpy to use it and hide the mechanics of the copy / IO flags.

Reply:
This code pattern is only met in two places, so I decided not to extract it in a helper function. We may do that in the future when it's more widespread. WDYT?